starky2/src/mock_stark.rs

```diff
@@ -0,0 +1,362 @@
+    const NUM_MULTIPLICANDS: usize,
+> {
+    num_rows: usize,
+    _phantom: PhantomData<F>,
+}
+
+impl<F: RichField + Extendable<D>, const D: usize, const W: usize> MultiplicationStark<F, D, W> {
+    fn multiplicand(&self, i: usize) -> usize {
+        debug_assert!(i < W);
+        i
+    }
+
+    // Product of the first `i` multiplicands
+    fn intermediate_product(&self, i: usize) -> usize {
+        debug_assert!(i < W && i > 0);
+        W + i - 1
+    }
+
+    fn product(&self) -> usize {
+        2 * W - 2
+    }
+
+    const fn num_columns() -> usize {
+        2 * W - 1
+    }
+
+    fn new(num_rows: usize) -> Self {
+        Self {
+            num_rows,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn generate_trace(&self, multiplicands: &[Vec<F>]) -> Vec<PolynomialValues<F>>
+    where
+        [(); Self::num_columns()]:,
+    {
+        debug_assert_eq!(multiplicands.len(), self.num_rows);
+        let mut trace_rows = multiplicands
+            .iter()
+            .map(|row| {
+                debug_assert_eq!(row.len(), W);
+                let mut result = [F::ZERO; Self::num_columns()];
+                for i in 0..W {
+                    result[self.multiplicand(i)] = row[i];
+                }
+                let mut acc = row[0] * row[1];
+                for i in 1..W - 1 {
+                    result[self.intermediate_product(i)] = acc;
+                    acc *= row[i + 1];
+                }
+                result[self.product()] = acc;
+                result
+            })
+            .collect::<Vec<_>>();
+        trace_rows_to_poly_values(trace_rows)
+    }
+}
+
+impl<F: RichField + Extendable<D>, const D: usize, const W: usize> Stark<F, D>
+    for MultiplicationStark<F, D, W>
+{
+    const COLUMNS: usize = 2 * W - 1;
+    const PUBLIC_INPUTS: usize = 0;
+
+    fn eval_packed_generic<FE, P, const D2: usize>(
+        &self,
+        vars: StarkEvaluationVars<FE, P>,
+        yield_constr: &mut ConstraintConsumer<P>,
+    ) where
+        FE: FieldExtension<D2, BaseField = F>,
+        P: PackedField<Scalar = FE>,
+    {
+        yield_constr.constraint(
+            vars.local_values[self.intermediate_product(1)]
+                - vars.local_values[self.multiplicand(0)] * vars.local_values[self.multiplicand(1)],
+        );
+        for i in 2..W - 1 {
+            yield_constr.constraint(
+                vars.local_values[self.intermediate_product(i)]
+                    - vars.local_values[self.intermediate_product(i - 1)]
+                        * vars.local_values[self.multiplicand(i)],
+            )
+        }
+    }
+
+    fn eval_ext_recursively(
+        &self,
+        builder: &mut CircuitBuilder<F, D>,
+        vars: StarkEvaluationTargets<D, { Self::COLUMNS }, { Self::PUBLIC_INPUTS }>,
+        yield_constr: &mut RecursiveConstraintConsumer<F, D>,
+    ) {
+        todo!()
+    }
+
+    fn constraint_degree(&self) -> usize {
+        2
+    }
+}
+
+#[cfg(test)]
```

Comment:
Seems like the whole `mock_stark` module should probably be `#[cfg(test)]`, or in examples or what not

Reply:
I removed `mock_stark`.